tests/executorch/runtime/test_modeling.py

```diff
@@ -0,0 +1,212 @@
+
+import os
+import tempfile
+import unittest
+
+import pytest
+from executorch.extension.pybindings.portable_lib import ExecuTorchModule
+from transformers import AutoTokenizer
+from transformers.testing_utils import (
+    require_read_token,
+    slow,
+)
+
+from optimum.executorchruntime import ExecuTorchModelForCausalLM
+
+
+class ExecuTorchModelIntegrationTest(unittest.TestCase):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    @slow
+    @pytest.mark.run_slow
+    def test_load_model_from_hub(self):
+        model = ExecuTorchModelForCausalLM.from_pretrained(
+            model_name_or_path="NousResearch/Llama-3.2-1B",
+            export=True,
+            task="text-generation",
+            recipe="xnnpack",
+        )
+        self.assertIsInstance(model, ExecuTorchModelForCausalLM)
+        self.assertIsInstance(model.model, ExecuTorchModule)
+
+    @slow
+    @pytest.mark.run_slow
+    def test_load_model_from_local_path(self):
+        from optimum.exporters.executorch import main_export
+
+        model_id = "NousResearch/Llama-3.2-1B"
+        task = "text-generation"
+        recipe = "xnnpack"
+
+        with tempfile.TemporaryDirectory() as tempdir:
+            # Export to a local dir
+            main_export(
+                model_name_or_path=model_id,
+                task=task,
+                recipe=recipe,
+                output_dir=tempdir,
+            )
+            self.assertTrue(os.path.exists(f"{tempdir}/model.pte"))
+
+            # Load the exported model from a local dir
+            model = ExecuTorchModelForCausalLM.from_pretrained(
+                model_name_or_path=tempdir,
+                export=False,
+            )
+            self.assertIsInstance(model, ExecuTorchModelForCausalLM)
+            self.assertIsInstance(model.model, ExecuTorchModule)
+
+    @slow
+    @pytest.mark.run_slow
+    def test_llama3_2_1b_text_generation_with_xnnpack(self):
+        model_id = "NousResearch/Llama-3.2-1B"
+        model = ExecuTorchModelForCausalLM.from_pretrained(
+            model_name_or_path=model_id,
+            export=True,
+            task="text-generation",
+            recipe="xnnpack",
+        )
+        self.assertIsInstance(model, ExecuTorchModelForCausalLM)
+        self.assertIsInstance(model.model, ExecuTorchModule)
+
+        EXPECTED_GENERATED_TEXT = "Simply put, the theory of relativity states that the laws of physics are the same in all inertial frames of reference."
+        tokenizer = AutoTokenizer.from_pretrained(model_id)
+        generated_text = model.text_generation(
+            tokenizer=tokenizer,
+            prompt="Simply put, the theory of relativity states that",
+            max_seq_len=len(tokenizer.encode(EXPECTED_GENERATED_TEXT)),
+        )
+        self.assertEqual(generated_text, EXPECTED_GENERATED_TEXT)
+
+    @slow
+    @pytest.mark.run_slow
+    def test_llama3_2_3b_text_generation_with_xnnpack(self):
```

Comment:
@michaelbenayoun If there is no M1 runner with larger RAM, we can skip this test for 3B model